src/internal_events/batch.rs

```diff
@@ -0,0 +1,24 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct LargeEventDropped {
+    pub length: usize,
+    pub max_length: usize,
+}
+
+impl InternalEvent for LargeEventDropped {
+    fn emit_logs(&self) {
+        error!(
+            message = "Event larger than batch max_bytes; dropping event.",
+            batch_max_bytes = %self.max_length,
+            length = %self.length,
+            internal_log_rate_secs = 1
+        );
+    }
+
+    fn emit_metrics(&self) {
+        counter!("events_discarded_total", 1);
+        counter!("large_events_total", 1);
```

Comment:
I'm not a huge fan of this name, but can't think of anything to replace it with at the moment...

Reply:
😅 I'll blame the late hour for me, that's a good name.

Reply:
`oversize_events_total` does sound better.